manim/utils/tex_file_writing.py

```diff
@@ -42,67 +45,88 @@ def generate_tex_file(expression, tex_template, source_type):
     result = Path(result).as_posix()
     tex_file = Path(tex_file).as_posix()
     tex_dir = Path(file_writer_config["tex_dir"]).as_posix()
     if not os.path.exists(result):
-        commands = (
-            [
-                "latex",
-                "-interaction=batchmode",
-                "-halt-on-error",
-                '-output-directory="{}"'.format(tex_dir),
-                '"{}"'.format(tex_file),
-                ">",
-                os.devnull,
-            ]
-            if not use_ctex
-            else [
-                "xelatex",
-                "-no-pdf",
-                "-interaction=batchmode",
-                "-halt-on-error",
-                '-output-directory="{}"'.format(tex_dir),
-                '"{}"'.format(tex_file),
-                ">",
-                os.devnull,
-            ]
-        )
-        exit_code = os.system(" ".join(commands))
+        command = tex_compilation_command(tex_compiler, tex_file, tex_dir)
+        exit_code = os.system(command)
         if exit_code != 0:
             log_file = tex_file.replace(".tex", ".log")
-            raise Exception(
-                (
-                    "LaTeX error converting to dvi. "
-                    if not use_ctex
-                    else "XeLaTeX error converting to xdv. "
-                )
-                + f"See log output above or the log file: {log_file}"
+            raise ValueError(
+                f"{tex_compiler['command']} error converting to"
+                f" {tex_compiler['output_format'][1:]}. See log output above or"
+                f" the log file: {log_file}"
             )
     return result
 
 
-def dvi_to_svg(dvi_file, use_ctex=False, regen_if_exists=False):
+def dvi_to_svg(
+    dvi_file, use_ctex=False, tex_compiler=None, regen_if_exists=False, page=1
+):
     """
-    Converts a dvi, which potentially has multiple slides, into a
-    directory full of enumerated pngs corresponding with these slides.
-    Returns a list of PIL Image objects for these images sorted as they
-    where in the dvi
+    Converts a dvi, xdv, or pdf file into an svg using dvisvgm.
     """
-    result = dvi_file.replace(".dvi" if not use_ctex else ".xdv", ".svg")
+    if tex_compiler is None:
+        tex_compiler = (
+            {"command": "xelatex", "output_format": ".xdv"}
+            if use_ctex
+            else {"command": "latex", "output_format": ".dvi"}
+        )
+    result = dvi_file.replace(tex_compiler["output_format"], ".svg")
     result = Path(result).as_posix()
     dvi_file = Path(dvi_file).as_posix()
     if not os.path.exists(result):
         commands = [
             "dvisvgm",
+            "--pdf" if tex_compiler["output_format"] == ".pdf" else "",
+            "-p " + str(page),
             '"{}"'.format(dvi_file),
             "-n",
-            "-v",
-            "0",
-            "-o",
-            '"{}"'.format(result),
+            "-v 0",
+            "-o " + '"{}"'.format(result),
```

Comment:
suggestion: Using fstrings make it more readable.
```suggestion
            "-o " + f'"{result}"',
```